src/PdoStreamIterator.php

```diff
@@ -120,10 +121,19 @@ public function current(): ?Message
             $metadata['_position'] = $this->currentItem->no;
         }
 
+        $payload = json_decode($this->currentItem->payload, true);
+        if (json_last_error() !== JSON_ERROR_NONE) {
```

Comment:
unit test should be simple, just insert some garbage to the database in a test-case